lib/ansible/modules/system/known_hosts.py

```diff
@@ -294,6 +297,30 @@ def normalize_known_hosts_key(key):
         d['key']=k[2]
     return d
 
+def compute_diff(path, found_line, replace_or_add, state, key):
+    diff = {
+        'before_header': path,
+        'after_header': path,
+        'before': '',
+        'after': '',
+    }
+    try:
+        inf = open(path, "r")
+    except IOError:
+        e = get_exception()
+        if e.errno == errno.ENOENT:
+            diff['before_header'] = '/dev/null'
+    else:
+        diff['before'] = inf.read()
+        inf.close()
+    diff['after'] = ''.join([
+        line for line_number, line in enumerate(diff['before'].splitlines(1))
+        if not (found_line==(line_number + 1) and (replace_or_add or state=='absent'))
```

Comment:
Note: found_line is 1-based, so it should be `del old_lines[found_line - 1]`.

I'm reluctant to do refactorings without unit tests.  I'll see if I can manage to write one.